packages/compiler/core/parser.ts

```diff
@@ -1,6 +1,20 @@
 import { createSymbolTable } from "./binder.js";
 import { codePointBefore, isIdentifierContinue } from "./charcode.js";
 import { compilerAssert } from "./diagnostics.js";
+import {
+  ArrayExpressionNode,
+  IntersectionExpressionNode,
+  ProjectionArithmeticExpressionNode,
+  ProjectionCallExpressionNode,
+  ProjectionDecoratorReferenceExpressionNode,
+  ProjectionEqualityExpressionNode,
+  ProjectionLogicalExpressionNode,
+  ProjectionMemberExpressionNode,
+  ProjectionRelationalExpressionNode,
+  ProjectionUnaryExpressionNode,
+  ReturnExpressionNode,
+  UnionExpressionNode,
+} from "./index.js";
```

Comment:
Was it intentional and necessary to move these from being imported from types to imported from index?

Reply:
feel like this is a bad pattern to import a parent index.js. You can endup with some circular import that cause things to not be loaded yet. (Might not be the case with es6 module anymore though)